### PR TITLE
Desktop Notifications

### PR DESCRIPTION
**Pull request by UmairMukhtar**
image

### CODE REVIEW

**Comment by UmairMukhtar** [comment]
[Read Notifications and view notifciations.webm](https://user-images.githubusercontent.com/34018917/229607495-5406db60-aa5c-44f8-8e07-7b6739235bf8.webm)

File: app/api/src/db/pool.ts [hunk 1]
```diff
@@ -92,6 +98,20 @@ const init = async (): Promise<QueryResult<any>> => {
           REFERENCES users(user_id)
           ON DELETE CASCADE
     );
+    
+    CREATE TABLE IF NOT EXISTS notification (
+      notification_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+      notification_for UUID NOT NULL,
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/db/pool.ts:104]: done

File: app/api/src/controllers/notification.controller.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  res.send(newNotification);
+});
+
+export const deleteNotificationById = catchAsync(async (req, res): Promise<void> => {
+  await notificationService.deleteNotificationById(req.params.notification_id);
+  res.send();
+});
+
+export const updateNotificationById = catchAsync(async (req, res): Promise<void> => {
+  // check if reference docs exist
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/controllers/notification.controller.ts]: done

File: app/api/src/services/notification.service.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        (options.ascend_fields || []).length > 0 ? `${options.ascend_fields.map((x) => `${x} ASC`).join(', ')}` : '';
+      const descendOrder =
+        (options.descend_fields || []).length > 0 ? `${options.descend_fields.map((x) => `${x} DESC`).join(', ')}` : '';
+      const order =
+        (options.ascend_fields || options.descend_fields || []).length > 0
+          ? `ORDER BY ${ascendOrder} ${
+              (options.ascend_fields || []).length > 0 && (options.descend_fields || []).length > 0 ? ', ' : ''
+            } ${descendOrder}`
+          : '';
+          console.log('THis is the order',order)
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/services/notification.service.ts]: done

File: app/api/src/utils/getSupportedFileTypeFromLink.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export const isValidYoutubeID = async (youtubeID:string) => {
 }
 export const getSupportedFileTypeFromLink = async (url: string): Promise<TLinkValue | null> => {
   try {
+    return supportedMediaTypes.IMAGE
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/utils/getSupportedFileTypeFromLink.ts]: done

**Comment by UmairMukhtar** [comment]
[click side bar notifications.webm](https://user-images.githubusercontent.com/34018917/229818528-a5c31891-b275-43f7-972a-5b07f3ba8174.webm)

**Review by AltiMario — APPROVED** [review]

**Pull request merged**